lib/reek/context/code_context.rb

```diff
@@ -16,13 +15,16 @@ module Context
     #
     # :reek:TooManyMethods: { max_methods: 19 }
     # :reek:TooManyInstanceVariables: { max_instance_variables: 8 }
+    # :reek:Attribute
     class CodeContext
       extend Forwardable
       delegate each_node: :exp
       delegate %i(name type) => :exp
-      delegate %i(visibility visibility= non_public_visibility?) => :visibility_tracker
 
-      attr_reader :children, :context, :exp, :statement_counter, :visibility_tracker
+      attr_accessor :visibility
```

Comment:
So we are back to ....just a value. Using an object and composition + delegation feels cleaner to me. But maybe we dont need that anymore! I'll just keep on reading ;)